doc/manual/src/architecture/file-system-object.md

```diff
@@ -0,0 +1,60 @@
+# File System Object
+
+Nix uses a simplified model of the file system, which consists of file system objects.
+Every file system object is one of the following:
+
+ - File
+
+   - A possibly empty sequence of bytes for contents
+   - A single boolean representing the [executable](https://en.m.wikipedia.org/wiki/File-system_permissions#Permissions) permission
+
+ - Directory
+
+   Mapping of names to child file system objects
+
+ - [Symbolic link](https://en.m.wikipedia.org/wiki/Symbolic_link)
+
+   A file system path that may point anywhere
+
+We call a store object's outermost file system object the *root*.
+
+A bare file or symlink can be a root file system object.
```

Comment:
```suggestion
File systems objects and their children form a tree.
The unique file system object from which all others descend in a tree is called the "root".
(This is just the customary terminology for trees in general, not specific to Nix or file system objects.)
```

I deleted the second sentence because it is also specific to store objects. We're just introducing an inductive datatype at this point, there is no reason to think a non-directory couldn't be a root.

(Conversely, once we document store objects, we can make the point properly that any sort of FSO can be the root FSO of the store object, not just directories. There it actually makes sense.)

Reply:
Even better, remove all mentions of "root" because that's not an intrinsic property of a file system object anyway.



> there is no reason to think

love it

Reply:
OK